### DISCUSSION TITLE
[Windows] Hangs on accept()

### DISCUSSION THREAD

**Discussion by godlikepanos** [DaanDeMeyer/reproc#89] [Q&A]
Hi,

I have a program with 16 threads where each thread invokes one external process thousands of times. So each thread hammers `reproc_start()`, `reproc_wait()` and `reproc_destroy()`. Unfortunately, after a while I might observe a hang where all threads are stuck in `socketpair()` and more precisely in `accept()`. If I set `REPROC_REDIRECT_DISCARD` in all redirects then there is no hang. 

I'm not sure what could be wrong. There are only 16 processes running at the same time. My guess is that there is some issue with the cleanup and the parent process runs out of descriptors or something.

Any ideas?

**Comment by VODKA555** [comment]
这是来自QQ邮箱的假期自动回复邮件。您好，您的邮件我已收到，我会尽快处理，谢谢 Liao Jiaxi
